s4e-backend/src/main/java/pl/cyfronet/s4e/sync/SceneAcceptor.java

```diff
@@ -0,0 +1,48 @@
+package pl.cyfronet.s4e.sync;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import pl.cyfronet.s4e.sync.context.Context;
+import pl.cyfronet.s4e.sync.step.Step;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Component
+@RequiredArgsConstructor
+@Slf4j
+public class SceneAcceptor {
+    private final PipelineFactory pipelineFactory;
+
+    public void accept(String sceneKey) {
+        Context context = new Context(sceneKey);
+        try {
+            List<Step<Context, Error>> pipeline = pipelineFactory.build();
+            for (Step<Context, Error> step : pipeline) {
+                Error error = step.apply(context);
+                if (error != null) {
+                    handleError(error);
+                    break;
+                }
+            }
+        } catch (RuntimeException e) {
+            handleError(context.getError().cause(e).build());
+        }
+    }
+
+    private void handleError(Error error) {
+        StringBuilder sb = new StringBuilder();
```

Comment:
There are mixed levels of abstraction, it's not necessary to be fixed. Abstraction are levels of encapsulation of methods, in if you used method and at the beginning, you created string builder. 

Example how can it be splited

function getErrorPrefixMsg(error) return new StringBuilder().append(...)...

function appendErrorCode(builder, error) => return error.getCode() != null ? builder.append(...)... : builder;

function appendErrorParams(builder, error) => return ....

Finally, something like this can be archived

handleError(error) {
builder = getErrorPrefixMsg(error)
appendErrorCode(builder, error)
appendErrorParams(builder, error)

log.warn(builder.toString(), ...)
}

Reply:
Sure, it could be split into separate methods. I will do that.